sdk/attestation/Azure.Security.Attestation/src/AttestationClient.cs

```diff
@@ -150,11 +164,13 @@ private async Task<AttestationResponse<AttestationResult>> AttestSgxEnclaveInter
                 {
                     response = _restClient.AttestSgxEnclave(attestSgxEnclaveRequest, cancellationToken);
                 }
-                var attestationToken = new AttestationToken(response.Value.Token);
-
+                var attestationToken = AttestationToken.Deserialize(response.Value.Token, _clientDiagnostics);
                 if (_options.TokenOptions.ValidateToken)
                 {
-                    await attestationToken.ValidateTokenInternalAsync(_options.TokenOptions, await GetSignersAsync(cancellationToken).ConfigureAwait(false), async, cancellationToken).ConfigureAwait(false);
+                    if (!await attestationToken.ValidateTokenInternal(_options.TokenOptions, await GetSignersAsync(cancellationToken).ConfigureAwait(false), async, cancellationToken).ConfigureAwait(false))
```

Comment:
Please see the token validation comment referencing sync/async concerns in `AttestationAdministrationClient`.   We may need to apply the same considerations here.